File: backend/src/main/java/com/astayc/hontodare/Service/CharacterService.java
package com.astayc.hontodare.Service;

import com.astayc.hontodare.DTO.CharacterDTO;
import com.astayc.hontodare.Entity.Character;
import java.util.List;
import java.util.UUID;

public interface CharacterService {
    List<CharacterDTO> getAllCharacters();
    CharacterDTO createCharacter(CharacterDTO characterDTO);
    List<CharacterDTO> getCharactersByCategory(String category);
    CharacterDTO updateCharacter(CharacterDTO characterDTO);
    void deleteCharacter(UUID id);
}

File: frontend/src/app/core/services/character.service.ts
// src/app/core/services/character.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import {Character} from "../models/character.model";

@Injectable({
  providedIn: 'root'
})
export class CharacterService {
  private apiUrl = `${environment.apiUrl}/api/character`;

  constructor(private http: HttpClient) {}

  uploadCharacter(formData: FormData): Observable<any> {
    return this.http.post(`${this.apiUrl}/upload`, formData);
  }

  getAllCharacters(): Observable<Character[]> {
    return this.http.get<Character[]>(this.apiUrl);
  }

  getCharacterById(id: number): Observable<Character> {
    return this.http.get<Character>(`${this.apiUrl}/${id}`);
  }

  getCharactersByCategory(category: string): Observable<Character[]> {
    return this.http.get<Character[]>(`${this.apiUrl}/category/${category}`);
  }

  createCharacter(formData: FormData): Observable<Character> {
    return this.http.post<Character>(`${this.apiUrl}/upload`, formData);
  }

  updateCharacter(id: number, character: Character): Observable<Character> {
    return this.http.put<Character>(`${this.apiUrl}/${id}`, character);
  }

  deleteCharacter(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
